File: challenges/ttl-store/go/starter/exercise.go
package exercise

import "refresher/tooling/go/contracts"

// Add the state required by the challenge contract.
type TTLStore struct{}

func NewTTLStore(clock contracts.Clock) *TTLStore {
	panic("Not implemented")
}

func (s *TTLStore) Put(key, value string, ttlMS int64) {
	panic("Not implemented")
}

func (s *TTLStore) Get(key string) (string, bool) {
	panic("Not implemented")
}

func (s *TTLStore) Delete(key string) bool {
	panic("Not implemented")
}

File: challenges/ttl-store/php/starter/exercise.php
<?php
declare(strict_types=1);

namespace Refresher;

use Closure;
use LogicException;

final class TTLStore implements KeyValueStore
{
    public function __construct(Closure $clock) {
        throw new LogicException("Not implemented");
    }

    public function put(string $key, string $value, int $ttlMs): void
    {
        throw new LogicException("Not implemented");
    }

    public function get(string $key): ?string
    {
        throw new LogicException("Not implemented");
    }

    public function delete(string $key): bool
    {
        throw new LogicException("Not implemented");
    }
}
